### ISSUE TITLE
meta: Reth's built-in tracers

### ISSUE THREAD

**Issue by jxom** [foundry-rs/foundry#5855]
### Component

Anvil

### Describe the feature you would like

The requirement is pretty straight-forward: would be nice to have support for Reth's built-in tracers (namely `callTracer` & `prestateTracer`) in Anvil. This would help us build an intuitive UI inside [Rivet](https://github.com/paradigmxyz/rivet) for tracing & state changes on transactions (and calls).

It was [noted in a previous issue](https://github.com/foundry-rs/foundry/issues/4110#issuecomment-1575919746) that there may be consolidation efforts to bring native tracers over from Reth and into Anvil, and an option (for the meantime) was also raised to just copy-paste the Reth native tracer implementation into Anvil for the time being. I am happy to give the copy-paste part a shot if the consolidation efforts are not to happen anytime soon, but if it is, I am happy to wait!

Related:

- [x] https://github.com/foundry-rs/foundry/issues/9569
- [x] https://github.com/foundry-rs/foundry/issues/8776
- [x] https://github.com/foundry-rs/foundry/issues/6764

**Comment by mattsse** [comment]
this is definitely planned and only blocked by type unification which is currently in progress

**Comment by metaclips** [comment]
Hello, any update on this?

**Comment by figtracer** [comment]
Both requested built-in tracers, `callTracer` and `prestateTracer`, are now [implemented in Anvil](https://github.com/foundry-rs/foundry/blob/92594bda71533db5eb8978d39a80319ffde7fcd8/crates/anvil/src/eth/backend/mem/mod.rs#L6515-L6586).
